lib/elixir/lib/integer.ex

```diff
@@ -11,9 +11,35 @@ defmodule Integer do
   Returns `true` if `n` is an odd number, otherwise `false`.
 
   Allowed in guard clauses.
+
+  ## Examples
+
+      iex> Integer.is_odd(3)
+      true
+
+      iex> Integer.is_odd(4)
+      false
   """
+  @spec is_odd(integer) :: boolean
   defmacro is_odd(n) do
-    quote do: (unquote(n) &&& 1) == 1
+    case Macro.Env.in_guard?(__CALLER__) do
+      true ->
+        quote do: (unquote(n) &&& 1) == 1
+
+      false ->
+        quote bind_quoted: [n: n] do
```

Comment:
We should never generate this much quoted code in the user. I don't think we should fix this, let's continue generating badarg. It sucks but at least it is consistent with all guard like things.